### DISCUSSION TITLE
.mjs files doesnt' support?

### DISCUSSION THREAD

**Discussion by stc1988** [Moddable-OpenSource/moddable#1066] [Q&A] [answered]
I added `.mjs` file to modules section of `manifest.json` ,but this file didn't compiled to xsb file. 
According to  [mcmanifest.js](https://github.com/Moddable-OpenSource/moddable/blob/public/tools/mcmanifest.js#L1154), it doesn't support `.mjs` files.

Is this behavior an issue?

**Comment by phoddie** [comment, marked as answer]
> it doesn't support .mjs files...

Correct. `.mjs` is a Node.js anachronism to assist in the transition from CommonJS to standard ECMAScript modules. XS in the Moddable SDK has long supported only standard ECMAScript modules so this seemed unnecessary.

> Is this behavior an issue?

It has not been to date. But, you are asking the question for a reason. Is this an issue for you?

**Reply by phoddie** [reply]
FYI – the work on `mcpack` provided a motivation for supporting `.mjs` source files. So, you should be able to use those now.

**Comment by stc1988** [comment]
Thank you for replying. I understand why XS doesn't support `.mjs`.

> you are asking the question for a reason. Is this an issue for you?

I'm trying to use some npm packages from the Moddable SDK experimentally, but some of these are distributed as `.mjs` files. Of course it works if I rename the file to `.js`.

**Comment by phoddie** [comment]
> I'm trying to use some npm packages from the Moddable SDK experimentally, but some of these are distributed as .mjs files. Of course it works if I rename the file to .js.

Thanks for explaining. That makes sense. If necessary, I don't see a problem adding .mjs support

**Reply by jermsam** [reply]
There is a package called to-esm that converts commonjs modules to mjs modules, hopefully by adding support for .mjs, this entire non compatibility issue will be solved

**Reply by phoddie** [reply]
`mcconfig` and `mcrun` now handle `.mjs` files. This [change](https://github.com/Moddable-OpenSource/moddable/commit/7da4085376e9a8ca37b8dfb68ecbdc8bac951c73) was made in August of this year as part of the early work on `mcpack`. I just confirmed that it works by renaming `$MODDABLE/examples/helloworld/main.js` to `main.mjs` and performing a clean build:

```
cd $MODDABLE/examples/helloworld
mcconfig -d -m -t clean
mcconfig -d -m
```

**Reply by jermsam** [reply]
for my proposed solution to work, to-esm must first successfully convert all sub dependencies  to .mjs which is not always possible
